webapp/src/js/enketo/widgets/rdtoolkit-provision.js

```diff
@@ -0,0 +1,146 @@
+{
+  'use strict';
+  const Widget = require('enketo-core/src/js/Widget');
+  const $ = require('jquery');
+  const utils = require('../widget-utils');
+  require('enketo-core/src/js/plugins');
+
+  const pluginName = 'rdtoolkitprovisionwidget';
+
+  /**
+   * @constructor
+   * @param {Element} element [description]
+   * @param {(boolean|{touch: boolean, repeat: boolean})} options options
+   */
+  function Rdtoolkitprovisionwidget(element, options) {
+    this.namespace = pluginName;
+    Widget.call(this, element, options);
+    this._init();
+  }
+
+  // Copy the prototype functions from the Widget super class
+  Rdtoolkitprovisionwidget.prototype = Object.create(Widget.prototype);
+
+  // Ensure the constructor is the new one
+  Rdtoolkitprovisionwidget.prototype.constructor = Rdtoolkitprovisionwidget;
+
+  Rdtoolkitprovisionwidget.prototype.destroy = function(element) {};  // eslint-disable-line no-unused-vars
+
+  Rdtoolkitprovisionwidget.prototype._init = function() {
+    const $widget = $(this.element);
+    const rdToolkitService = window.CHTCore.RDToolkit;
+
+    if (!rdToolkitService.enabled()) {
+      window.CHTCore.Translate
+        .get('rdtoolkit.disabled')
+        .toPromise()
+        .then(label => $widget.append(`<label>${label}</label>`));
+      return;
+    }
+
+    displayActions($widget);
+
+    $widget.on('click', '.btn.rdtoolkit-provision-test', () => provisionRDTest($widget, rdToolkitService));
+  };
+
+  function provisionRDTest($widget, rdToolkitService) {
+    const dateFormat = 'LLL';
+    const form = utils.getForm();
+    // Using form's instance ID as RD Test ID
+    const sessionId = utils.getFieldValue(form, 'rdtoolkit-provision > meta > instanceID').replace('uuid:', '');
+    const patientId = utils.getFieldValue(form, 'rdtoolkit-provision > patient_id');
```

Comment:
Same question: if we hardcode the model root, does it mean we can only use it with one form?

Reply:
Done!